mlflow/xgboost/__init__.py

```diff
@@ -663,3 +483,232 @@ def infer_model_signature(input_example):
 
     safe_patch(FLAVOR_NAME, xgboost, "train", train, manage_run=True)
     safe_patch(FLAVOR_NAME, xgboost.DMatrix, "__init__", __init__)
+
+    # enable xgboost scikit-learn estimators autologging
+    import mlflow.sklearn
+
+    flavor_name_xgb_sklearn = "xgboost_sklearn"
+    xgb_sklearn_autolog = autologging_integration(flavor_name_xgb_sklearn)(mlflow.sklearn._autolog)
+    xgb_sklearn_autolog(
+        flavor_name=flavor_name_xgb_sklearn,
+        log_input_examples=log_input_examples,
+        log_model_signatures=log_model_signatures,
+        log_models=log_models,
+        disable=disable,
+        exclusive=exclusive,
+        disable_for_unsupported_versions=disable_for_unsupported_versions,
+        silent=silent,
+        max_tuning_runs=max_tuning_runs,
+        log_post_training_metrics=True,
+    )
+
+
+def _mlflow_xgboost_logging(
```

Comment:
If we adopt the suggestion from https://github.com/mlflow/mlflow/pull/5078/files#diff-fc3d121bec0c47e040232d0fec24356f7f4d0423d1bfee58bfa3dbb9dc1b79ebR1273 to only call XGBoost MLflow logging methods for XGBoost scikit-learn estimators, we can remove this method and put it's contents back in the `train()` method defined in `autolog()`.